code/modules/mob/living/carbon/xenomorph/mutators/strains/drone/healer.dm

```diff
@@ -173,3 +183,118 @@
 /datum/behavior_delegate/drone_healer/proc/un_salve()
 	salve_applied_recently = FALSE
 	bound_xeno.update_icons()
+
+/*
+	SACRIFICE
+*/
+
+/datum/behavior_delegate/drone_healer/proc/modify_transferred(amount)
+	transferred_amount += amount
+
+/datum/behavior_delegate/drone_healer/append_to_stat()
+	. = list()
+	. += "Transferred health amount: [transferred_amount]/[required_transferred_amount]"
+	if(transferred_amount >= required_transferred_amount)
+		. += "Sacrifice will grant you new life."
+
+/datum/behavior_delegate/drone_healer/on_life()
+	if(!bound_xeno)
+		return
+	if(bound_xeno.stat == DEAD)
+		return
+	var/image/holder = bound_xeno.hud_list[PLASMA_HUD]
+	holder.overlays.Cut()
+	var/percentage_transferred = min(round((transferred_amount / required_transferred_amount) * 100, 10), 100)
+	if(percentage_transferred)
+		holder.overlays += image('icons/mob/hud/hud.dmi', "xenoenergy[percentage_transferred]")
+
+/datum/behavior_delegate/drone_healer/handle_death(mob/M)
+	var/image/holder = bound_xeno.hud_list[PLASMA_HUD]
+	holder.overlays.Cut()
+
+/datum/action/xeno_action/activable/healer_sacrifice
+	name = "Sacrifice"
+	action_icon_state = "screech"
+	ability_name = "sacrifice"
+	var/max_range = 1
+	var/transfer_mod = 0.75 // only transfers 75% of current healer's health
+	macro_path = /datum/action/xeno_action/verb/verb_healer_sacrifice
+	action_type = XENO_ACTION_CLICK
+	ability_primacy = XENO_PRIMARY_ACTION_5
+
+/datum/action/xeno_action/verb/verb_healer_sacrifice()
+	set category = "Alien"
+	set name = "Sacrifice"
+	set hidden = TRUE
+	var/action_name = "Sacrifice"
+	handle_xeno_macro(src, action_name)
+
+/datum/action/xeno_action/activable/healer_sacrifice/use_ability(atom/atom)
+	var/mob/living/carbon/xenomorph/xeno = owner
+	var/mob/living/carbon/xenomorph/target = atom
+
+	if(!istype(target))
+		return
+
+	if(target == xeno)
+		to_chat(xeno, "You can't heal yourself!")
+		return
+
+	if(isfacehugger(target) || islesserdrone(target))
+		to_chat(xeno, "It would be a waste...")
+		return
+
+	if(!xeno.check_state())
+		return
+
+	if(target.stat == DEAD)
+		to_chat(xeno, SPAN_WARNING("[target] is already dead!"))
+		return
+
+	if(target.health >= target.maxHealth)
+		to_chat(xeno, SPAN_WARNING("[target] is already at max health!"))
+		return
+
+	if(!isturf(xeno.loc))
+		to_chat(xeno, SPAN_WARNING("You cannot transfer health from here!"))
+		return
+
+	if(get_dist(xeno, target) > max_range)
+		to_chat(xeno, SPAN_WARNING("You need to be closer to [target]."))
+		return
```

Comment:
It is currently possible to sacrifice yourself to enemy hives. Consider adding something along the lines of:
https://github.com/cmss13-devs/cmss13/blob/657862c08c2c376df7b23aa3480ac7264093b94c/code/modules/mob/living/carbon/xenomorph/mutators/strains/drone/healer.dm#L106-L108